test/powershell/Modules/Microsoft.PowerShell.Utility/Implicit.Remoting.Tests.ps1

```diff
@@ -1,2120 +1,2105 @@
-﻿Describe "Implicit remoting and CIM cmdlets with AllSigned and Restricted policy" -tags "Feature" {
-
-    BeforeAll {
-
-        # Skip test for non-windows machines
-        $skipTest = !$IsWindows
-
-        if ($skipTest) { return }
-
-        #
-        # GET CERTIFICATE
-        #
-
-        $tempName = "$env:TEMP\signedscript_$(Get-Random).ps1"
-        "123456" > $tempName
-        $cert = $null
-        foreach ($thisCertificate in (Get-ChildItem cert:\ -rec -codesigning))
-        {
-	        $null = Set-AuthenticodeSignature $tempName -Certificate $thisCertificate
-	        if ((Get-AuthenticodeSignature $tempName).Status -eq "Valid")
-	        {
-		        $cert = $thisCertificate
-		        break
-	        }
-        }
-
-        # Skip the tests if we couldn't find a code sign certificate
-        # This will happen in NanoServer and IoT
-        if ($null -eq $cert)
-        {
-            $skipTest = $true
-            return
-        }
-
-        # Ensure the cert is trusted
-        if (-not (Test-Path "cert:\currentuser\TrustedPublisher\$($cert.Thumbprint)"))
-        {
-            $store = New-Object System.Security.Cryptography.X509Certificates.X509Store "TrustedPublisher"
-            $store.Open("ReadWrite")
-            $store.Add($cert)
-            $store.Close()
-        }
-
-        #
-        # Set process scope execution policy to 'AllSigned'
-        #
-
-        $oldExecutionPolicy = Get-ExecutionPolicy -Scope Process
-        Set-ExecutionPolicy AllSigned -Scope Process
-
-        #
-        # Create a remote session
-        #
-
-        $session = New-RemoteSession
-    }
-
-    AfterAll {
-        if ($skipTest) { return }
-
-        if ($null -ne $tempName) { Remove-Item -Path $tempName -Force -ErrorAction SilentlyContinue }
-        if ($null -ne $oldExecutionPolicy) { Set-ExecutionPolicy $oldExecutionPolicy -Scope Process }
-        if ($null -ne $session) { Remove-PSSession $session -ErrorAction SilentlyContinue }
-    }
-
+﻿#
+# Skip all tests on non-windows and non-PowerShellCore and non-elevated platforms.
+#
+$originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+$originalWarningPreference = $WarningPreference
+$WarningPreference = "SilentlyContinue"
+$skipTest = ! ($IsWindows -and $IsCoreCLR -and (Test-IsElevated))
```

Comment:
Most of these tests should work over PSRP/SSH, right?  Is there an issue already to enable our remoting tests over SSH?

Reply:
Yes #3904.  We may be able to selectively run existing remoting tests on WinRM or SSH after we have a way to automatically install and configure SSH on various platforms (#3616).